arelle/plugin/validate/DBA/rules/fr.py

```diff
@@ -94,6 +97,41 @@ def rule_fr7(
     )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_fr9(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    DBA.FR9: The year's result (fsa:ProfitLoss) must be filled in as part of the income statement.
+    The control only looks at instances without dimensions or instances that only have the dimension
+    (ConsolidatedSoloDimension with ConsolidatedMember).
+
+    Implementation: Find any occurrence of fsa:ProfitLoss with no dimensions or with dimensions of
+    ConsolidatedSoloDimension with ConsolidatedMember.  If nothing is found, trigger an error.
+    """
+    profLossFacts = val.modelXbrl.factsByQname.get(pluginData.profitLossQn)
+    found = False
+    if profLossFacts:
+        for fact in profLossFacts:
+            if (fact.context is not None
+                    and fact.context.qnameDims.get(PluginValidationDataExtension.consolidatedSoloDimensionQn) == PluginValidationDataExtension.consolidatedMemberQn
```

Comment:
I'd suggest using `fact.context.dimMemberQname` so we retrieve the implied/default value and also don't rely on the context's internal data structure for dimension values.